record-builder-core/src/main/java/io/soabase/recordbuilder/core/RecordBuilder.java

```diff
@@ -267,6 +267,16 @@
          */
         String beanClassName() default "";
 
+        /**
+         * When to add the {@link javax.annotation.processing.Generated} annotation to generated files
+         *
+         * <p>
+         * If set to 'ALWAYS' (default), then always add the {@link javax.annotation.processing.Generated} annotation to
+         * generated files.
+         * </p>
+         */
+        GeneratedAnnotation addGeneratedAnnotation() default GeneratedAnnotation.ALWAYS;
```

Comment:
Let's use a boolean here instead.

Reply:
So, not giving a choice for later options, like `IF_ON_CLASSPATH` or `USE_USER_SUPPLIED_ANNOTATION`? Just want to check. Booleans are restrictive, no choice of extending the options later on!

Reply:
We already have an option for "if on classpath" (that should be respected, BTW) so that ship has already sailed. Also, the use of the enum is very cumbersome. Let's use a boolean please.

Reply:
I haven't seen such an option. How can I enabled the "Generated" annotation already? It's not on the classpath and it got generated.
If we could make that working, let's just drop by this PR and issue.

Reply:
@Randgalt I have looked for this option but did not find it.